dss/stepfunctions/visitation/storage.py

```diff
@@ -103,3 +133,90 @@ def _aggregate(self, work_results: Sequence[Mapping[str, List[int]]]) -> Mutable
                     for i, delta in enumerate(deltas):
                         counters[i] += delta
         return aggregate
+
+
+class KeyInfo(NamedTuple):
+    """
+    Encapsulates everything we can know about a key in a replica's storage bucket.
+    """
+    bucket: str
+    key: str
+    user_metadata: Mapping[str, str]
+    size: int
+    content_type: str
+    cloud_checksum: str
+
+
+class StorageFolder(ABC):
```

Comment:
What's the point of using ABCs here? Is it to just auto-register subclasses in `cls_by_name`? I think it would be better accomplished by simply listing the classes in an array or something like that.

Reply:
I forgot to actually declare inspect_key() as abstract.

The registration stuff is unrelated. I'll take it out and replace it with an explicit list. It would be more LOCs but less magic.

Reply:
Fixed.